File: nowcoder/question/NC156.cpp
class Solution {
public:
    /**
     * 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
     *
     *
     * @param arr intvector
     * @param k int
     * @return int
     */
    int foundOnceNumber(vector<int>& arr, int k) {
        // write code here
        sort(arr.begin(), arr.end());
        for (int i = 0; i < arr.size() - 1; ++i) {
            if (arr[i] == arr[i + 1]) {
                i += k - 1;
            }
            else {
                return arr[i];
            }
        }
        return *(arr.end() - 1);
    }
};

File: 常规遍历/nowcoder/NC 156.js
function foundOnceNumber( arr ,  k ) {
    // write code here
    arr.sort()
    let index = 0
    while(index < arr.length){
        if(index != arr.length - 1 && arr[index] == arr[index + 1]){
            index += k
        }else{
            return arr[index]
        }
    }
}